src/components/NewDogForm.jsx

```diff
@@ -0,0 +1,63 @@
+import { useState } from 'react';
+import PropTypes from 'prop-types';
+import './NewDogForm.css';
+
+const NewDogForm = ({ addDogCallback }) => {
+  const [dogData, setDogData] = useState({
+    name: '',
+    breed: '',
+    age: '',
+  });
+
+  const submitDogData = (e) => {
+    e.preventDefault();
+
+    addDogCallback(dogData);
+    setDogData({ name: '', breed: '', age: '' });
+  };
+
+  const handleChange = (e) => {
+    setDogData({ ...dogData, [e.target.name]: e.target.value });
+  };
+
+  return (
+    <form onSubmit={submitDogData} className="new-dog__form">
+      <section>
+        <h2>Add a Dog</h2>
+        <div className="new-dog__fields">
+          <label htmlFor="name">Name</label>
```

Comment:
Students might ask us to show the `useId` hook that's now mentioned in the curriculum, so how about a comment to remind us that's a thing now and that we should review the use of `useId` in preparation.